src/Sylius/Component/Core/Distributor/MinimumPriceDistributor.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Component\Core\Distributor;
+
+use Sylius\Component\Core\Model\ChannelInterface;
+use Sylius\Component\Core\Model\OrderItemInterface;
+use Sylius\Component\Core\Model\ProductVariantInterface;
+
+final class MinimumPriceDistributor implements MinimumPriceDistributorInterface
+{
+    private ProportionalIntegerDistributorInterface $proportionalIntegerDistributor;
+
+    public function __construct(ProportionalIntegerDistributorInterface $proportionalIntegerDistributor)
+    {
+        $this->proportionalIntegerDistributor = $proportionalIntegerDistributor;
+    }
+
+    public function distribute(array $orderItems, int $amount, ChannelInterface $channel): array
+    {
+        $orderItemsToProcess = [];
+        foreach ($orderItems as $index => $orderItem) {
+            /** @var ProductVariantInterface $variant */
+            $variant = $orderItem->getVariant();
+
+            $minimumPrice = $variant->getChannelPricingForChannel($channel)->getMinimumPrice();
+
+            $minimumPrice *= $orderItem->getQuantity();
+
+            $orderItemsToProcess['order-item-' . $index] = [
+                'orderItem' => $orderItem,
+                'minimumPrice' => $minimumPrice,
+            ];
+        }
+
+        return array_values(array_map(
+            function (array $processedOrderItem): int { return $processedOrderItem['promotion']; },
```

Comment:
probably we could use arrow functions here and below

Reply:
yeah, we could use it here but it could be done in other PR as part of refactor.